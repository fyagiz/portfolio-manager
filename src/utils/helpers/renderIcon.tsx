import { Ionicons, MaterialCommunityIcons } from "@expo/vector-icons";
import { RenderIconType } from "./helpers.type";

const renderIcon = (renderIconSpecs: RenderIconType) => {
  const { iconType, iconName, color, size } = renderIconSpecs;
  switch (iconType) {
    case "Ionicons":
      return <Ionicons name={iconName} color={color} size={size} />;
    case "MaterialCommunityIcons":
      return <MaterialCommunityIcons name={iconName} color={color} size={size} />;
    default:
      return <MaterialCommunityIcons name="map-marker-question" size={24} color="black" />;
  }
};

export default renderIcon;
