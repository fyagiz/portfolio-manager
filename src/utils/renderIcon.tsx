import { Ionicons, MaterialCommunityIcons } from "@expo/vector-icons";
import { RenderIconType } from "./utilTypes";

export const renderIcon = (renderIconSpecs: RenderIconType) => {
  const { iconType, iconName, color, size } = renderIconSpecs;
  switch (iconType) {
    case "Ionicons":
      return <Ionicons name={iconName} color={color} size={size} />;
    case "MaterialCommunityIcons":
      return <MaterialCommunityIcons name={iconName} color={color} size={size} />;
    default:
      break;
  }
};
