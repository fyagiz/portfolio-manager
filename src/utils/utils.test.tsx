import { render } from "@testing-library/react-native";
import { renderIcon } from "./renderIcon";

describe("Utils Unit Tests", () => {
  test("should renderIcon render Ionicons icons", () => {
    const icon = render(renderIcon({ iconType: "Ionicons", color: "red", iconName: "add", size: 24 }));
    expect(icon).toMatchSnapshot();
  });
  test("should renderIcon render MaterialCommunityIcons icons", () => {
    const icon = render(renderIcon({ iconType: "MaterialCommunityIcons", color: "red", iconName: "ab-testing", size: 24 }));
    expect(icon).toMatchSnapshot();
  });
  test("should renderIcon render unknown icons", () => {
    // To test default branch.
    // @ts-expect-error
    const icon = render(renderIcon({ iconType: "UnknownIcon", color: "red", iconName: "ab-testing", size: 24 }));
    expect(icon).toMatchSnapshot();
  });
});
