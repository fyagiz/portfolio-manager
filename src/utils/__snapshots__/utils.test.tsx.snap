// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Utils Unit Tests should renderIcon render Ionicons icons 1`] = `<Text />`;

exports[`Utils Unit Tests should renderIcon render MaterialCommunityIcons icons 1`] = `<Text />`;

exports[`Utils Unit Tests should renderIcon render unknown icons 1`] = `
<Text
  allowFontScaling={false}
  selectable={false}
  style={
    [
      {
        "color": "black",
        "fontSize": 24,
      },
      undefined,
      {
        "fontFamily": "material-community",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      {},
    ]
  }
>
  ó°¼‡
</Text>
`;
